### ISSUE TITLE
Use of RawDecoderSettings

### ISSUE THREAD

**Issue by davidekholm** [lanthale/LibRawFX#3]
I can't see how to make use of RawDecoderSettings. Is its usage not implemented yet?

**Comment by lanthale** [comment]
I have not finished the implementation. Actually I had no time to implement it. Therefore everyone is invited to add this functionality.
Actual I am waiting for the JDK19 release because there are many fixes included which is directly influence the Windows. 
Plan:
- Upgrade to JDK19
- Upgrade Linxu to a newer libc version
- Upgrade code base to use new features of panama
- Implement RawDecoderSettings

If want you can help me regarding the decodersetting part

**Comment by davidekholm** [comment]
Are there unresolved issues with the Foreign Memory and Linking API?

**Comment by lanthale** [comment]
In JDK18 there was an issue where under Windows and in combination with Jpackage native launcher Panama was not working anymore (crashing the JVM). This is now solved at least for single threaded access but if you have more native libs to access in parallel the second lib is not working but not crashing the JVM. 
In JDK19 all of the above issues are resolved but I have to re-generate the code with jextract.

**Comment by davidekholm** [comment]
Thank you for that info. We're currently using the Advanced Installer launcher on Windows so we're hopefully not affected. Perhaps another JDK 18 update fixes this too?

**Comment by lanthale** [comment]
I have just checked with JDK19 and the issue is fixed. But there is another issue in JDK19 added which pervents the lib working on Apple M1 architecture. Checkout the new version for JDK19

**Comment by davidekholm** [comment]
Thank you! Does this update mean that I can adjust RAW settings while reading images now, for instance brightness and white balance?

**Comment by lanthale** [comment]
No not yet. Wait until the final version of jdk 19 is released and I will update with the options you need. Before I have no time to work on LibrawFX. 

Cheers
Clemens

**Comment by davidekholm** [comment]
Thanks for reply. Do you think they've fixed what's broken with the arm64 version then too or is that postponed for a future v19 release?

(When all is working, I plan on writing an ImageIO wrapper around your library for our jAlbum software)

**Comment by lanthale** [comment]
This will be fixed in the first versions of JDK20. A pull request is already created and waiting for inclusion into jdk20. I have no information if the fix will be backported to jdk19 but there is a chance because it is effecting many topics on arm64 (stack allocation is different). 

Great to hear that you plan to integrate into jalbum. 

Cheers 
Clemens

**Comment by davidekholm** [comment]
Hi again. Just chiming in to know whether I can now adjust RAW settings while reading images now, for instance brightness and white balance?
